lib/schema.js

```diff
@@ -78,6 +78,18 @@ Schema.prototype.validate = function(data) {
     }
   }
 
+  if(this.schema.strict || (opts && opts.strict)) {
```

Comment:
thinking more on this, i think the example you bring up is a good one and likely also the main one to focus on. the idea of turning 'strictness' is an all or nothing point of view (all validation is strict, or none of it is)

with that in mind, (and also based on your test cases) how about we remove the check for `this.schema.strict` and have strictness be something that gets passed in as an option to validate only.

it still feels a bit hidden, but having it be an option object, the key name itself at least is clear.

you okay with that?